tests/utils/test_date_handler.py

```diff
@@ -47,3 +47,17 @@ def test_get_previous_month_dates_if_random_month_of_the_year(self):
         self.assertTupleEqual(
             get_date_start_and_date_stop_from_date_range("PREVIOUS_MONTH"), (date(2021, 1, 1), date(2021, 1, 31))
         )
+
+    @parameterized.expand(
+        [
+            (None, date(2021, 1, 12), None),
+            (None, date(2021, 1, 12), "YESTERDAY"),
+            (date(2021, 1, 12), None, None),
+            (date(2021, 1, 12), None, "YESTERDAY"),
+            (None, None, None),
+            (date(2021, 1, 12), date(2021, 1, 31), "YESTERDAY"),
+        ]
+    )
+    def test_check_date_range_definition_conformity(self, start_date, end_date, date_range):
+        with self.assertRaises(Exception):
+            check_date_range_definition_conformity(start_date, end_date, date_range)
```

Comment:
You also need to test the case where the dates are correctly defined. Just check that the function returns `None`.